cni/cmd/istio-cni/main.go

```diff
@@ -44,3 +50,24 @@ func main() {
 	skel.PluginMain(plugin.CmdAdd, plugin.CmdCheck, plugin.CmdDelete, version.All,
 		fmt.Sprintf("CNI plugin istio-cni %v", istioversion.Info.Version))
 }
+
+// runUnshare triggers subcommand of istio-cni. This is used to invoke ourself to avoid dependencies on the node (the *one*
+// thing we know must exist on the node is this binary, since its called from ourselves!).
+// This will prepare the mount namespace(s) before executing a command.
+func runUnshare() {
+	fs := flag.NewFlagSet("unshare", flag.ExitOnError)
+	netns := fs.String("lock-file", "", "file to override lock")
+	if err := fs.Parse(os.Args[2:]); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	if len(fs.Args()) < 1 {
+		fmt.Fprint(os.Stderr, "usage: unshare --lock-file=file -- <command> [args...]\n")
+		os.Exit(1)
+	}
+	// Skip first arg which is "unshare"
+	if err := mount.RunSandboxed(*netns, fs.Args()); err != nil {
```

Comment:
Why sandbox the entire `istio-cni` process versus just the `iptables` process, which is what needs the workarounds?

Can we just wrap the iptables exec in this and move it into our iptables helper lib?

Reply:
To be clear, we call istio-cni many times now:
* Once, the normal top level call
* For each iptables call: `istio-cni unshare iptables ...`

'istio-cni unshare' basically is `sh -c mount .... ; exec iptables ...` but without dependencies.